unittests/test_parameters.py

```diff
@@ -0,0 +1,150 @@
+# Copyright 2016-2020 Swiss National Supercomputing Centre (CSCS/ETH Zurich)
+# ReFrame Project Developers. See the top-level LICENSE file for details.
+#
+# SPDX-License-Identifier: BSD-3-Clause
+
+from unittests.resources.checks.paramcheck import (NoParams, TwoParams,
+                                                   Abstract, ExtendParams)
+
+
+def test_param_space_is_empty():
+    class MyTest(NoParams):
+        pass
+
+    assert MyTest._rfm_params == {}
+
+
+def test_params_are_present():
+    class MyTest(TwoParams):
+        pass
+
+    assert MyTest._rfm_params['P0'] == ('a',)
+    assert MyTest._rfm_params['P1'] == ('b',)
```

Comment:
Here I wouldn't look into the `_rfm_params`, since this is an implementation detail. I would rather register normally the tests that I'm interested in and then check that the `P0`, `P1` are defined correctly. This way, you will also check that the test registration happens correctly.

Reply:
Such test is also further down. My intention with this test was to check only the logic related to the parameter space build, and make sure that the parameter overriding, inheritance and filtering works. 
Building `cls._rfm_params` and populating the actual parameters in the object are two different steps and I think they should be tested separately. The build of `cls._rfm_params` is dealt with by the logic implemented into the metaclass, but populating the actual parameters into the object is done by the pipeline.

How about adding to `RegressionTest` a method such as 
```python
@classmethod
def get_parameter_space(cls):
    return cls._rfm_params
```
This would allow us to test that the parameter space is built correctly without the need of using any implementation details.